<?xml version="1.0"?>
<launch>

  <arg name="model" default="$(find edymobile)/urdf/edymobile_theo.urdf.xacro"/>
  <arg name="rvizconfig" default="$(find edymobile)/rviz/edy_world.rviz" />
  
  
  <arg name="map_file" value="/$(find edymobile)/maps/circuit.yaml">
  </arg> 
  <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_file)"> 
  	<param name="frame_id" value="map" />
  </node> 
  
  <!-- Transformation Configuration ... Setting Up the Relationships Between Coordinate Frames --> 
  <node pkg="tf" type="static_transform_publisher" name="base_link_broadcaster" args="0 0 0.053 0 0 0 base_link base_center 30" />
  <!--node pkg="tf" type="static_transform_publisher" name="imu_broadcaster" args="-0.0315 0 0.033 0 0 0 base_link imu 30" /-->
  <!-- odom to base_footprint transform will be provided by the robot_pose_ekf node -->
  <node pkg="tf" type="static_transform_publisher" name="map_to_odom_broadcaster" args="0 0 0 0 0 0 map odom 30" />

  <include file="$(find edymobile)/launch/gazebo_swisscat_world.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="rvizconfig" value="$(arg rvizconfig)" />
  </include>


  <rosparam command="load"
            file="$(find edymobile)/config/joints.yaml"
            ns="edy_joint_state_controller" />

  
  <!-- Wheel Odometry Publisher -->
  <!-- Subscribe: /right_ticks, /left_ticks, /initial_2d -->
  <!-- Publish: /odom_data_euler, /odom_data_quat -->
  <node pkg="localization_data_pub" type="ekf_odom_pub" name="ekf_odom_pub">
  </node> 
	
  <!-- Extended Kalman Filter from robot_pose_ekf Node-->
  <!-- Subscribe: /odom, /imu_data, /vo -->
  <!-- Publish: /robot_pose_ekf/odom_combined -->
  <!--remap from="scan" to="scan_filtered" /-->
  <remap from="odom" to="odom_data_quat" />
  <remap from="imu_data" to="imu_normalized" />
  <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    <param name="output_frame" value="odom"/>
    <param name="base_footprint_frame" value="base_link"/>
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="false"/>
    <param name="vo_used" value="false"/>
    <param name="gps_used" value="false"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="false"/>
  </node>
	
  <!-- Initial Pose and Goal Publisher -->
  <!-- Publish: /initialpose, /move_base_simple/goal -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)">
  <!--node pkg="rviz" type="rviz" name="rviz"-->
  </node> 

  <!-- Subscribe: /initialpose, /move_base_simple/goal -->
  <!-- Publish: /initial_2d, /goal_2d --> 
  <node pkg="localization_data_pub" type="rviz_click_to_2d" name="rviz_click_to_2d">
  </node>   

  <node name="rqt_robot_steering" pkg="rqt_robot_steering" type="rqt_robot_steering">
    <param name="default_topic" value="/edymobile/cmd_vel"/>
  </node>

  <!-- Move Base Node -->
  <!-- Subscribe: /move_base_simple/goal -->
  <!-- Publish: /cmd_vel -->
  <!--
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find edymobile)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find edymobile)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find edymobile)/param/local_costmap_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find edymobile)/param/global_costmap_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find edymobile)/param/base_local_planner_params.yaml" command="load" />
  </node>
  -->

</launch>
